detox/src/devices/drivers/android/genycloud/services/GenyRecipesService.js

```diff
@@ -0,0 +1,30 @@
+const Recipe = require('./dto/GenyRecipe');
+
+class GenyRecipesService {
+  constructor(genyCloudExec, logger) {
+    this.genyCloudExec = genyCloudExec;
+    this.logger = logger;
+  }
+
+  async getRecipeByName(recipeName) {
+    const result = await this.genyCloudExec.getRecipe(recipeName);
+    const { recipes } = result;
+    if (!recipes.length) {
+      return null;
+    }
+
+    if (recipes.length > 1) {
+      this.logger.warn({ event: 'GENYCLOUD_RECIPE_LOOKUP'}, `More than one Genymotion-Cloud recipe found for recipe name ${recipeName}`);
+    }
+    return new Recipe(recipes[0]);
+  }
+
+  async getRecipeByUUID(uuid) {
```

Comment:
Looks like a fake. If this class cannot fulfill this function, then why do you pretend like you can do it here?
I'd suggest wrapping the UUID inside `GenyDeviceQueryHelper` class, to be as straightforward as possible here.

In other words, **this** is a recipe:

```json
{"uuid": "107d757e-463a-4a18-8667-b8dec6e4c87e", "name": "Google Pixel", "android_version": "9.0", "screen_width": 1080, "screen_height": 1920, "screen_density": 420, "screen": "1080 x 1920 dpi 420", "source": "genymotion"}
```

And **this** is just a mere stub object:

```
{ uuid, name: `Recipe of ${uuid}` }
```